tensorflow-nn/src/TensorFlow/NN.hs

```diff
@@ -73,13 +73,12 @@ sigmoidCrossEntropyWithLogits
   :: (OneOf '[Float, Double] a, TensorType a, Num a)
      => Tensor Value a          -- ^ __logits__
      -> Tensor Value a          -- ^ __targets__
-     -> Build (Tensor Value a)
+     -> Tensor Value a
 sigmoidCrossEntropyWithLogits logits targets = do
     let zeros = zerosLike logits
         cond = logits `greaterEqual` zeros
         relu_logits = select cond logits zeros
         neg_abs_logits = select cond (-logits) logits
-    withNameScope "logistic_loss" $ do
```

Comment:
@fkm3, could you take a look at why Build is not behaving here? I know we eventually should have a non-Build story for assigning scopes, but until then, it would be nice to not have to choose between good names and correct code :)